src/Jobs/ReindexAfterWriteJob.php

```diff
@@ -61,32 +61,42 @@ public function getJobType()
         return QueuedJob::QUEUED;
     }
 
-
     /**
      * Lets process
      */
     public function process()
     {
-        if ($this->id && $this->class) {
-            $service = Injector::inst()->get('Heyday\Elastica\ElasticaService');
-            $reading_mode = Versioned::get_reading_mode();
-            Versioned::set_reading_mode('Stage.Live');
-
-            $versionToIndex = DataObject::get($this->class)->byID($this->id);
-            if (!$versionToIndex->hasField('ShowInSearch') || $versionToIndex->ShowInSearch) {
-                $service->index($versionToIndex);
-            } else {
-                $service->remove($versionToIndex);
-            }
 
-            $this->updateDependentClasses($versionToIndex, $service);
+        if (!$this->id || !$this->class) {
+            // No valid data to index
+            return;
+        }
+
+        $service = Injector::inst()->get('Heyday\Elastica\ElasticaService');
```

Comment:
Could you use `ElasticaService::class` here with a use statement at the top instead of the string?

Reply:
@jakxnz Keen to get this change in 👍